src/models/user.ts

```diff
@@ -0,0 +1,9 @@
+import { Document, Model, model } from 'mongoose'
+
+import { IUser } from '../interfaces'
+import { schemaNames, UserSchema } from '../schema'
+
+export interface IUserModel extends IUser, Document {
```

Comment:
why this is empty?

Reply:
IUser is our custom interface and to provide the functionality of save, remove, update. We must extend it with the document. That's why i have just extends it and there is not done anything else.